react/Product.js

```diff
@@ -5,19 +5,45 @@ import PropTypes from 'prop-types'
 // eslint-disable-next-line no-restricted-imports
 import { pathOr, path, sort, last, flatten } from 'ramda'
 import { jsonLdScriptProps } from 'react-schemaorg'
+import { useQuery } from 'react-apollo'
+import GET_SETTINGS from './queries/getSettings.graphql'
 
 const getSpotPrice = path(['commertialOffer', 'spotPrice'])
 const getPrice = path(['commertialOffer', 'Price'])
+const getTax = path(['commertialOffer', 'Tax'])
 const getAvailableQuantity = pathOr(0, ['commertialOffer', 'AvailableQuantity'])
+const DEFAULT_DECIMALS = 2
+const DEFAULT_PRICES_WITH_TAX = false
+
+const getAppSettings = () => {
+  const { data } = useQuery(GET_SETTINGS, { ssr: false })
+
+  if (!data) return {decimals: DEFAULT_DECIMALS, pricesWithTax: DEFAULT_PRICES_WITH_TAX}
+
+  const { decimals, pricesWithTax } = JSON.parse(data.appSettings.message)
+
+  return {
+    decimals: decimals,
+    pricesWithTax: pricesWithTax
+  }
+}
+
+const getFinalPrice = (value, decimals, pricesWithTax, getPriceFunc) => {
+  return pricesWithTax ? Math.round(((getPriceFunc(value) + getTax(value)) + Number.EPSILON) * Math.pow(10, decimals)) / Math.pow(10, decimals) : getPriceFunc(value)
+}
 
 const sortByPriceAsc = sort(
   (itemA, itemB) => getSpotPrice(itemA) - getSpotPrice(itemB)
 )
 
+const sortByPriceWithTaxAsc = sort(
+  (itemA, itemB) => (getSpotPrice(itemA) + getTax(itemA)) - (getSpotPrice(itemB) + getTax(itemB))
+)
+
 const isSkuAvailable = (sku) => getAvailableQuantity(sku) > 0
 
-const lowHighForSellers = (sellers) => {
-  const sortedByPrice = sortByPriceAsc(sellers)
+const lowHighForSellers = (sellers, pricesWithTax) => {
```

Comment:
Add the second parameter as object, it will improve maintainability. Also type this object

```suggestion
interface LowHighForSellersOptions {
  ...
}
const lowHighForSellers = (sellers, { pricesWithTax }: LowHighForSellersOptions) => {
```

Reply:
Done, added the object but since it's a JS file, interfaces are not allowed. Only allowed in tsx files AFAIK

Reply:
Yeah, my bad, I didn't see that's a js file.